### PR TITLE
:sparkles: Sign the inventory client for the global hub agent

### PR DESCRIPTION
**Pull request by yanmxa**
## Related issue(s)

Fixes # https://issues.redhat.com/browse/ACM-12395

## Tests
* [x] Unit/function tests have been added and incorporated into `make unit-tests`.
* [ ] Integration tests have been added and incorporated into `make integration-test`.
* [ ] E2E tests have been added and incorporated into `make e2e-test-all`.
* [x] List other manual tests you have done.
   - use the sample inventory component to test it manually

### CODE REVIEW

**Review by clyang82 — APPROVED** [review]
LGTM

File: pkg/transport/rest_credentail.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package transport
+
+import "sigs.k8s.io/kustomize/kyaml/yaml"
+
+type RestfulConnCredentail struct {
```
Comment (clyang82) [inline pkg/transport/rest_credentail.go:5]: suggest to use `RestfulConnConfig`

File: pkg/transport/kafka_credentail.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package transport
+
+import "sigs.k8s.io/kustomize/kyaml/yaml"
+
+// KafkaConnCredential is used to connect the transporter instance. The field is persisted to secret
+// need to be encode with base64.StdEncoding.EncodeToString
+type KafkaConnCredential struct {
```
Comment (clyang82) [inline pkg/transport/kafka_credentail.go:7]: suggest to use `KafkaConfig`

**Comment by clyang82** [comment]
/lgtm

will have a following PR to rename the struct name.

File: pkg/transport/rest_credentail.go (hunk 1 above)
Comment (yanmxa) [reply pkg/transport/rest_credentail.go:5]: Change the naming issue in another pr

File: pkg/transport/kafka_credentail.go (hunk 2 above)
Comment (yanmxa) [reply pkg/transport/kafka_credentail.go:7]: Change the naming issue in another pr

**Pull request merged**